### PR TITLE
feat(diffusers/pipelines): add unit test for pipelines

### PR DESCRIPTION
**Pull request by The-truthh**
# What does this PR do?

1. add fast unit test for pipelines
2. fix some bugs
  (1) Fix the bug of converting the dtype of `weight` will results in an error when weight is `None` in `groupnorm`.
  (2) Replace `ops.pad` with `ms.mint.nn.functional.pad` to address the precision issue.
  (3) Fix the bug where the output is a tensor when output_type is set to np in deepfloyd_if.
  (4) Modify the dtype of `watermark_image` initialization in watermark to maintain consistency with diffusers in deepfloyd_if. 
  (5) Fix the bug in usage of  `tensor.flatten` in stable_diffedit.
  (6) Ensure the backbone in DPT model is initialized with `set_train=False` to maintain consistency with diffusers.
  (7) Manually broadcasting the shape of `attn_mask` to ensure proper execution of the `FlashAttentionScore`

# Usage

1. Test a single pipeline
pytest tests/diffusers/pipelines/xxx/test_xxx_pipeline.py

2. Test all pipelines
pytest tests/diffusers/pipelines/

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: mindone/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -730,6 +730,9 @@ def flash_attention_op(
         # In MindSpore, False indicates retention and True indicates discard, in PyTorch it is the opposite
         if attn_mask is not None:
             attn_mask = ops.logical_not(attn_mask) if attn_mask.dtype == ms.bool_ else attn_mask.bool()
+            attn_mask = ops.broadcast_to(
```
Comment (CaitinZhao) [inline mindone/diffusers/models/attention_processor.py:733]: 看下有没有其他显存占用小一点的修改方式

File: mindone/diffusers/models/normalization.py [hunk 2]
```diff
@@ -468,7 +468,9 @@ def __init__(self, num_groups: int, num_channels: int, eps: float = 1e-5, affine
             self.bias = None
 
     def construct(self, x: Tensor):
-        x = group_norm(x, self.num_groups, self.weight.to(x.dtype), self.bias.to(x.dtype), self.eps)
+        weight = self.weight.to(x.dtype) if self.weight is not None else None
```
Comment (CaitinZhao) [inline mindone/diffusers/models/normalization.py:471]: 用 self.affine 控制

**Review by CaitinZhao — APPROVED** [review]
